### PR TITLE
Fix extractive predict - make output deterministic, use no_grad and raw_scores

### PR DESCRIPTION
**Pull request by salman1993**
Currently, the output includes a single sentence. This is because the variable `outputs` in predict is a 2-dim tensor. This PR fixes that.

Test script:
```
from extractive import ExtractiveSummarizer

fpath = '../models/distilroberta-epoch=3.ckpt'

model = ExtractiveSummarizer.load_from_checkpoint(fpath)


t = """
Paul: Please see the enclosed and call me if you
have any questions. You will note  that I have
assumed that the waiver period would expire either
on a date  certain or if certain events happened
before such date. Please let me know  if there are
other things that may cause the waiver period to
expire early. Carol
""".strip()

print(f"text: {t}\n\n")

summary_sents = model.predict(t, raw_scores=True)
print(summary_sents)
print("\n")

summary_sents2 = model.predict(t, raw_scores=True)
print(summary_sents2)
```

Output before PR:
```
{'Paul : Please see the enclosed and call me if you \n have any questions.': tensor([0.0175, 0.1614, 0.0674, 0.0946], grad_fn=<SelectBackward>)}

{'Paul : Please see the enclosed and call me if you \n have any questions.': tensor([0.0159, 0.1420, 0.121, 0.068], grad_fn=<SelectBackward>)}
```

Output after PR:
```
{'Paul : Please see the enclosed and call me if you \n have any questions.': 0.014968900009989738, 'You will note   that I have \n assumed that the waiver period would expire either \n on a date   certain or if certain events happened \n before such date.': 0.13204512000083923, 'Please let me know   if there are \n other things that may cause the waiver period to \n expire early.': 0.11174969375133514, 'Carol.': 0.058350831270217896}


{'Paul : Please see the enclosed and call me if you \n have any questions.': 0.014968900009989738, 'You will note   that I have \n assumed that the waiver period would expire either \n on a date   certain or if certain events happened \n before such date.': 0.13204512000083923, 'Please let me know   if there are \n other things that may cause the waiver period to \n expire early.': 0.11174969375133514, 'Carol.': 0.058350831270217896}
```

### CODE REVIEW

**Comment by salman1993** [comment]
There are 2 more issues:
1. `torch.no_grad` should be used in `predict` since there is no need to calculate gradients 
2. If we run the same input twice, we get different scores. I think this is because the model is not in `eval` model but I have not confirmed it yet.

**Comment by HHousen** [comment]
These changes look good. Merging.
